versioned_docs/version-4.4/building-extensions/plugins/ajax-plugin.md

```diff
@@ -0,0 +1,184 @@
+---
+title: Ajax Plugin for Ad Hoc Jobs
+sidebar_position: 4
+---
+# Introduction
+
+Imagine a Joomla site which handles bookings for events. On the day of the event the people putting out the chairs want to know how many people are coming, so they enter a URL on their browser which runs a job on the Joomla site and outputs the number of attendees.
+
+This is the sort of job which can be handled easily by using an Ajax plugin. Despite its name, an Ajax plugin isn't restricted to Ajax requests from javascript; you can use it to handle normal GET requests from a browser. 
+
+For ad hoc jobs like these, using an Ajax plugin is better than using a [Custom Script](../custom-script/index.md) because the functionality is supported by Joomla, and generally doesn't require rework between Joomla versions. In addition you can keep all your jobs together in one PHP file.
+
+This example Ajax plugin provides a utility to list the number of extensions on the Joomla instance, grouped by extension type. You enter a URL in your browser:
+
+```
+http://yourdomain/index.php?option=com_ajax&format=raw&plugin=extensions
+```
+
+and the utility responds with the counts:
+
+```
+component:45
+file:3
+language:9 
+…
+```
+
+# Writing an Ajax plugin
+To write a plugin which handles the URL
+
+```
+http://yourdomain/index.php?option=com_ajax&format=raw&plugin=extensions
+```
+
+you simply have to write a standard plugin which listens for the event `onAjaxExtensions` and returns the result as a string. That's all that's required! Here's the pseudo code:

```

Comment:
I would suggest not using the word `Extensions` for this instead use Job or function or method or what ever not used in the joomla context as something different to prevent to be confusing.

Reply:
On the call we agreed just to leave it as "ad hoc jobs". I changed it from using `Extensions` to `getExtensionTotals`